code/script.js

```diff
@@ -1,9 +1,201 @@
+  el.className = "star";
+
+  const inner = document.createElement("span");
+  inner.className = "inner";
+  inner.innerText = starmojis[getRandomInt(0, starmojis.length - 1)];
+  el.appendChild(inner);
+  
+  elWrapper.appendChild(el);
+
+  const rect = el.getBoundingClientRect();
+
+  const lifetime = getRandomArbitrary(2000, 3000);
+
+  el.style.setProperty("--lifetime", lifetime);
+
+  const star = {
+    el,
+    absolutePosition: { x: rect.left, y: rect.top },
+    position: { x: rect.left, y: rect.top },
+    velocity: { x: vx, y: vy },
+    mass: 0.1, //kg
+    radius: el.offsetWidth, // 1px = 1cm
+    restitution: -.7,
+    
+    lifetime,
+    direction: vx > 0 ? 1 : -1,
+
+    animating: true,
+
+    remove() {
+      this.animating = false;
+      this.el.parentNode.removeChild(this.el);
+    },
+
+    animate() {
+      const star = this;
+      let Fx =
+        -0.5 *
+        Cd *
+        A *
+        rho *
+        star.velocity.x *
+        star.velocity.x *
+        star.velocity.x /
+        Math.abs(star.velocity.x);
+      let Fy =
+        -0.5 *
+        Cd *
+        A *
+        rho *
+        star.velocity.y *
+        star.velocity.y *
+        star.velocity.y /
+        Math.abs(star.velocity.y);
+
+      Fx = isNaN(Fx) ? 0 : Fx;
+      Fy = isNaN(Fy) ? 0 : Fy;
+
+      // Calculate acceleration ( F = ma )
+      var ax = Fx / star.mass;
+      var ay = ag + Fy / star.mass;
+      // Integrate to get velocity
+      star.velocity.x += ax * frameRate;
+      star.velocity.y += ay * frameRate;
+
+      // Integrate to get position
+      star.position.x += star.velocity.x * frameRate * 100;
+      star.position.y += star.velocity.y * frameRate * 100;
+      
+      star.checkBounds();
+      star.update();
+    },
+    
+    checkBounds() {
+
+      if (star.position.y > height - star.radius) {
+        star.velocity.y *= star.restitution;
+        star.position.y = height - star.radius;
+      }
+      if (star.position.x > width - star.radius) {
+        star.velocity.x *= star.restitution;
+        star.position.x = width - star.radius;
+        star.direction = -1;
+      }
+      if (star.position.x < star.radius) {
+        star.velocity.x *= star.restitution;
+        star.position.x = star.radius;
+        star.direction = 1;
+      }
+
+    },
+
+    update() {
+      const relX = this.position.x - this.absolutePosition.x;
+      const relY = this.position.y - this.absolutePosition.y;
+
+      this.el.style.setProperty("--x", relX);
+      this.el.style.setProperty("--y", relY);
+      this.el.style.setProperty("--direction", this.direction);
+    }
+  };
+
+  setTimeout(() => {
+    star.remove();
+  }, lifetime);
+
+  return star;
+}
+
+
+function animationLoop() {
+  var i = stars.length;
+  while (i--) {
+    stars[i].animate();
+
+    if (!stars[i].animating) {
+      stars.splice(i, 1);
+    }
+  }
+
+  requestAnimationFrame(animationLoop);
+}
+
+animationLoop();
```

Comment:
Okey, so this is where and why the stars and button are triggered when the page is refreshed - it's invoked automatically when you visit the page and not by an action on the specific button. You can try to have an onclick on the button and call the function there, for example. Play around with it a bit or do some reading to understand the why's and how's but it's a really good job with the JavaScript overall. Some info here:
https://www.w3schools.com/jsref/event_onclick.asp